### PR TITLE
halt workflow instance if disabled and naturally triggered

### PR DESCRIPTION
**Pull request by honnix**
# Hey, I just made a Pull Request!

## Description

If the workflow has been disabled between retry and it is naturally triggered, we halt it.

## Motivation and Context

The hypothesis is in this case users would most likely want the instance to stop.

**Note, this is a breaking change.**

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [x] All tests pass
- [x] Code coverage check passes
- [x] Error handling is tested
- [x] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [ ] This PR has NO breaking change to public API
- [x] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [x] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/ExecutionDescriptionHandler.java [hunk 1]
```diff
@@ -104,19 +103,26 @@ public void transitionInto(RunState state) {
...
 
-    final Collection<String> errors = validator.validateWorkflow(workflow);
+    if (data.trigger().isPresent()) {
+      if (data.trigger().orElseThrow().equals(Trigger.natural()) &&
+          !workflowWithState.state().enabled().orElse(false)) {
+        throw new MissingRequiredPropertyException(format("%s is disabled, halting %s", workflowId, workflowInstance));
+      }
+    }
+
+    final Collection<String> errors = validator.validateWorkflow(workflowWithState.workflow());
```
Comment (narape) [inline styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/ExecutionDescriptionHandler.java]: You have `var workflow = workflowWithState.workflow();` on line 111 already

File: styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/ExecutionDescriptionHandler.java [hunk 2]
```diff
@@ -104,19 +103,26 @@ public void transitionInto(RunState state) {
...
+
+    var dockerImage = workflow.configuration().dockerImage().orElseThrow(
+        () -> new MissingRequiredPropertyException(
+            format("%s has no docker image, halting %s", workflowId, workflowInstance))
     );
 
-    final Collection<String> errors = validator.validateWorkflow(workflow);
+    if (data.trigger().isPresent()) {
+      if (data.trigger().orElseThrow().equals(Trigger.natural()) &&
+          !workflowWithState.state().enabled().orElse(false)) {
```
Comment (narape) [inline styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/ExecutionDescriptionHandler.java]: This is hard to follow. What about rewriting it in this way?

```java
var isWorkflowEnabled = workflowWithState.state().enabled().orElse(false);
var isNaturalTrigger = data.trigger().map(t -> t.equals(Trigger.natural()).orElse(false);

if (!isWorkflowEnabled && isNaturalTrigger) {
```

**Comment by honnix** [comment]
Will ship this change in a separate release.
